### DISCUSSION TITLE
Roborock S4 Max support

### DISCUSSION THREAD

**Discussion by ThatTallGuy21** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#278] [Q&A] [answered]
Hello -- Are there any plans to support the roborock.vacuum.a19 (i.e. S4 Max)? When I pass through my credentials, token, and host I get the following message within my camera entity. I can get my S5 to display. 

![FailedToRetrieveMap_S4Max](https://user-images.githubusercontent.com/86063203/165434761-ef27d4e4-01e1-4e79-aa5d-dbb011680228.PNG)

**Comment by PiotrMachowski** [comment, marked as answer]
It should work out of the box. Try restarting it

**Reply by ThatTallGuy21** [reply]
Unfortunately that didn't work. I have tried restarting HA. I also tried removing the camera yaml for my S4 Max, restarting HA, then adding the camera yaml back, and restarting again.

**Reply by ThatTallGuy21** [reply]
I was able to get it to work, but my setup is not ideal. Totally unrelated to your integration, which is great by the way! Just wanted to drop a note in case anyone else runs into my situation and totally open to advice. 

Unlike my S5, which is capable of being connected to my Guest network on a different subnet using either the token/IP or signing into my MiiHome account, the S4 Max connected to the Guest network had nothing but issues while using the standard Xiaomi integration in HA. It would be found, it just wouldn't work. The S4 Max was the reason I had to use the Miot Auto HACS integration. It wasn't until I used that, and ONLY while using my sign in credentials. The token method didn't work regardless of how man new ones I tried for the S4 Max. It was sign in credential method using my online account that was the only way I was able to interact with it in HA. 

When I setup your Map Extractor and the LoveLace card to display both my S5 and S4 Max, it worked for only the S5. Nothing I did fixed that, I reset the WIFI a couple times and used the new token in my YAML. The LoveLace card would find the S4 Max but it would either say it couldn't load or couldn't find the device. The logs showed constant errors of not being able to find the S4 Max and the Extractor Map setup was constantly timing out despite it still working with the HACS Miot Auto integration.

Once I added my S4 Max to my Primary Network, it displayed the map as expected. I just updated the YAML to the new token and IP address. I don't understand the difference between both my devices and why one will only work on the Primary network. The only thing I can think of is to try changing the IP address of the S4 Max and putting it back on the Guest network. But that's not easily done with my router so will take more trial and error of factory resetting my S4 Max. No idea why it only works on my Primary Network, which is not ideal and makes me a bit uneasy.
